pkg/events/derive/derive.go

```diff
@@ -10,6 +12,10 @@ import (
 // Use as static variable for testability reasons
 var getEventDefFunc = events.Definitions.Get
 
+// InvokedContext is a map used to store the triggering events' context
+var InvokedContext = make(map[uint64]trace.Event)
```

Comment:
I think we shouldn't leave this as a global variable, and now we have a global mutex too.
It could very simply be fit into a struct like the `procinfo` cache.